nifty/pipeline/steps/nifsSort.py

```diff
@@ -141,18 +142,22 @@ def start():
 
     # Download data from gemini public archive to ./rawData/.
     if program:
-        url = 'https://archive.gemini.edu/download/'+ str(program) + '/notengineering/NotFail/present/canonical'
         if not os.path.exists('./rawData'):
             os.mkdir('./rawData')
-        logging.info('\nDownloading data from Gemini public archive to ./rawData. This will take a few minutes.')
-        logging.info('\nURL used for the download: \n' + str(url))
-        if proprietaryCookie:
-            download_query_gemini(url, './rawData', proprietaryCookie)
+        if cadc:
+            logging.info('\nDownloading data from the CADC archive to ./rawData. This will take a few minutes.')
+            download_query_cadc(program, os.getcwd()+'/rawData')
         else:
-            download_query_gemini(url, './rawData')
+            url = 'https://archive.gemini.edu/download/'+ str(program) + '/notengineering/NotFail/present/canonical'
```

Comment:
Is putting this URL building information into the download_query_gemini  more appropriate now?  To mirror the download_query_cadc?   

For the CADC one, we reroute the user to the GSA if we don't have the file, because proprietary.  Passing the proprietary key into download_query_cadc would enable that.  (would require a try-except on the query, and a check to see if permission denied and was the file being asked for from Gemini, or the cookie could just get added to all requests).

Reply:
6b976a1 adds the URL building information into the download_query_gemini() function.

Hopefully I understand what you're looking for as actionables from your second point:
- In download_query_cadc(), if the file is not found with a CADC query (even if the user requested that), we'll retry with a proprietary Gemini download?